### ISSUE TITLE
'pkg.install' is not available.

### ISSUE THREAD

**Issue by SamuelGogarty** [saltstack/salt#49853]
I apologize if there is vital info missing from this bug report, as this is the first bug report I have filed.

When attempting to use the pkg.install module on FreeBSD 11.2-RELEASE with the line

`salt '*' pkg.install vim` it errors out with the following.

```
[INFO    ] User root Executing command pkg.install with jid 20181001141315539875
[DEBUG   ] Command details {'fun': 'pkg.install', 'arg': ['vim'], 'tgt': '*', 'jid': '20181001141315539875', 'ret': '', 'tgt_type': 'glob', 'user': 'root'}
[INFO    ] Starting a new job with PID 65817
[ERROR   ] Exception raised when processing __virtual__ function for salt.loader.127.0.0.1.int.module.kernelpkg_linux_yum.opt-1. Module will not be loaded: 'NoneType' object has no attribute 'lower'
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1722, in _process_virtual
    virtual = getattr(mod, virtual_func)()
  File "/usr/local/lib/python3.6/site-packages/salt/modules/kernelpkg_linux_yum.py", line 41, in __virtual__
    elif __grains__.get('os', '').lower() \
AttributeError: 'NoneType' object has no attribute 'lower'
[WARNING ] salt.loader.127.0.0.1.int.module.kernelpkg_linux_yum.opt-1.__virtual__() is wrongly returning `None`. It should either return `True`, `False` or a new name. If you're the developer of the module 'opt-1', please fix this.
[ERROR   ] Exception raised when processing __virtual__ function for salt.loader.127.0.0.1.int.module.kernelpkg_linux_yum. Module will not be loaded: 'NoneType' object has no attribute 'lower'
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1722, in _process_virtual
    virtual = getattr(mod, virtual_func)()
  File "/usr/local/lib/python3.6/site-packages/salt/modules/kernelpkg_linux_yum.py", line 41, in __virtual__
    elif __grains__.get('os', '').lower() \
AttributeError: 'NoneType' object has no attribute 'lower'
[WARNING ] salt.loader.127.0.0.1.int.module.kernelpkg_linux_yum.__virtual__() is wrongly returning `None`. It should either return `True`, `False` or a new name. If you're the developer of the module 'kernelpkg_linux_yum', please fix this.
[ERROR   ] Exception raised when processing __virtual__ function for salt.loader.127.0.0.1.int.module.xbpspkg.opt-1. Module will not be loaded: 'in <string>' requires string as left operand, not NoneType
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1722, in _process_virtual
    virtual = getattr(mod, virtual_func)()
  File "/usr/local/lib/python3.6/site-packages/salt/modules/xbpspkg.py", line 37, in __virtual__
    if __grains__['os'] in ('Void') and _check_xbps():
TypeError: 'in <string>' requires string as left operand, not NoneType
[WARNING ] salt.loader.127.0.0.1.int.module.xbpspkg.opt-1.__virtual__() is wrongly returning `None`. It should either return `True`, `False` or a new name. If you're the developer of the module 'opt-1', please fix this.
[ERROR   ] Exception raised when processing __virtual__ function for salt.loader.127.0.0.1.int.module.xbpspkg. Module will not be loaded: 'in <string>' requires string as left operand, not NoneType
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1722, in _process_virtual
    virtual = getattr(mod, virtual_func)()
  File "/usr/local/lib/python3.6/site-packages/salt/modules/xbpspkg.py", line 37, in __virtual__
    if __grains__['os'] in ('Void') and _check_xbps():
TypeError: 'in <string>' requires string as left operand, not NoneType
[WARNING ] salt.loader.127.0.0.1.int.module.xbpspkg.__virtual__() is wrongly returning `None`. It should either return `True`, `False` or a new name. If you're the developer of the module 'xbpspkg', please fix this.
[ERROR   ] Exception raised when processing __virtual__ function for salt.loader.127.0.0.1.int.module.bsd_shadow.opt-1. Module will not be loaded: argument of type 'NoneType' is not iterable
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1722, in _process_virtual
    virtual = getattr(mod, virtual_func)()
  File "/usr/local/lib/python3.6/site-packages/salt/modules/bsd_shadow.py", line 30, in __virtual__
    if 'BSD' in __grains__.get('os', ''):
TypeError: argument of type 'NoneType' is not iterable
[WARNING ] salt.loader.127.0.0.1.int.module.bsd_shadow.opt-1.__virtual__() is wrongly returning `None`. It should either return `True`, `False` or a new name. If you're the developer of the module 'opt-1', please fix this.
[ERROR   ] Exception raised when processing __virtual__ function for salt.loader.127.0.0.1.int.module.bsd_shadow. Module will not be loaded: argument of type 'NoneType' is not iterable
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1722, in _process_virtual
    virtual = getattr(mod, virtual_func)()
  File "/usr/local/lib/python3.6/site-packages/salt/modules/bsd_shadow.py", line 30, in __virtual__
    if 'BSD' in __grains__.get('os', ''):
TypeError: argument of type 'NoneType' is not iterable
[WARNING ] salt.loader.127.0.0.1.int.module.bsd_shadow.__virtual__() is wrongly returning `None`. It should either return `True`, `False` or a new name. If you're the developer of the module 'bsd_shadow', please fix this.
[ERROR   ] Failed to read the virtual function for module: opt-1
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1775, in _process_virtual
    virtual
  File "/usr/local/lib/python3.6/site-packages/salt/utils/versions.py", line 138, in warn_until
    salt_version=_version_.formatted_version
RuntimeError: The warning triggered on filename '/usr/local/lib/python3.6/site-packages/salt/loader.py', line number 1775, is supposed to be shown until version 2014.1.0 (Hydrogen) is released. Current version is now 2018.3.1 (Oxygen). Please remove the warning.
[ERROR   ] Failed to read the virtual function for module: opt-1
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1775, in _process_virtual
    virtual
  File "/usr/local/lib/python3.6/site-packages/salt/utils/versions.py", line 138, in warn_until
    salt_version=_version_.formatted_version
RuntimeError: The warning triggered on filename '/usr/local/lib/python3.6/site-packages/salt/loader.py', line number 1775, is supposed to be shown until version 2014.1.0 (Hydrogen) is released. Current version is now 2018.3.1 (Oxygen). Please remove the warning.
[ERROR   ] Failed to read the virtual function for module: opt-1
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1775, in _process_virtual
    virtual
  File "/usr/local/lib/python3.6/site-packages/salt/utils/versions.py", line 138, in warn_until
    salt_version=_version_.formatted_version
RuntimeError: The warning triggered on filename '/usr/local/lib/python3.6/site-packages/salt/loader.py', line number 1775, is supposed to be shown until version 2014.1.0 (Hydrogen) is released. Current version is now 2018.3.1 (Oxygen). Please remove the warning.
[ERROR   ] Failed to read the virtual function for module: opt-1
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1775, in _process_virtual
    virtual
  File "/usr/local/lib/python3.6/site-packages/salt/utils/versions.py", line 138, in warn_until
    salt_version=_version_.formatted_version
RuntimeError: The warning triggered on filename '/usr/local/lib/python3.6/site-packages/salt/loader.py', line number 1775, is supposed to be shown until version 2014.1.0 (Hydrogen) is released. Current version is now 2018.3.1 (Oxygen). Please remove the warning.
[ERROR   ] Failed to read the virtual function for module: opt-1
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1775, in _process_virtual
    virtual
  File "/usr/local/lib/python3.6/site-packages/salt/utils/versions.py", line 138, in warn_until
    salt_version=_version_.formatted_version
RuntimeError: The warning triggered on filename '/usr/local/lib/python3.6/site-packages/salt/loader.py', line number 1775, is supposed to be shown until version 2014.1.0 (Hydrogen) is released. Current version is now 2018.3.1 (Oxygen). Please remove the warning.
[INFO    ] User root Executing command saltutil.find_job with jid 20181001141320672515
[DEBUG   ] Command details {'fun': 'saltutil.find_job', 'arg': ['20181001141315539875'], 'tgt': ['GrueWorld'], 'jid': '20181001141320672515', 'ret': '', 'tgt_type': 'list', 'user': 'root'}
[INFO    ] Starting a new job with PID 65820
[DEBUG   ] LazyLoaded saltutil.find_job
[DEBUG   ] LazyLoaded direct_call.execute
[DEBUG   ] Minion return retry timer set to 9 seconds (randomized)
[INFO    ] Returning information for job: 20181001141320672515
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
[DEBUG   ] Trying to connect to: tcp://127.0.0.1:4506
[DEBUG   ] minion return: {'success': True, 'return': {'pid': 65817, 'fun': 'pkg.install', 'arg': ['vim'], 'tgt': '*', 'jid': '20181001141315539875', 'ret': '', 'tgt_type': 'glob', 'user': 'root'}, 'retcode': 0, 'jid': '20181001141320672515', 'fun': 'saltutil.find_job', 'fun_args': ['20181001141315539875']}
[ERROR   ] Failed to read the virtual function for module: opt-1
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1775, in _process_virtual
    virtual
  File "/usr/local/lib/python3.6/site-packages/salt/utils/versions.py", line 138, in warn_until
    salt_version=_version_.formatted_version
RuntimeError: The warning triggered on filename '/usr/local/lib/python3.6/site-packages/salt/loader.py', line number 1775, is supposed to be shown until version 2014.1.0 (Hydrogen) is released. Current version is now 2018.3.1 (Oxygen). Please remove the warning.
[DEBUG   ] LazyLoaded systemd.booted
[ERROR   ] Failed to read the virtual function for module: opt-1
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1775, in _process_virtual
    virtual
  File "/usr/local/lib/python3.6/site-packages/salt/utils/versions.py", line 138, in warn_until
    salt_version=_version_.formatted_version
RuntimeError: The warning triggered on filename '/usr/local/lib/python3.6/site-packages/salt/loader.py', line number 1775, is supposed to be shown until version 2014.1.0 (Hydrogen) is released. Current version is now 2018.3.1 (Oxygen). Please remove the warning.
[DEBUG   ] SMBIOS: neither dmidecode nor smbios found!
[DEBUG   ] SMBIOS: neither dmidecode nor smbios found!
[ERROR   ] Failed to read the virtual function for module: opt-1
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1775, in _process_virtual
    virtual
  File "/usr/local/lib/python3.6/site-packages/salt/utils/versions.py", line 138, in warn_until
    salt_version=_version_.formatted_version
RuntimeError: The warning triggered on filename '/usr/local/lib/python3.6/site-packages/salt/loader.py', line number 1775, is supposed to be shown until version 2014.1.0 (Hydrogen) is released. Current version is now 2018.3.1 (Oxygen). Please remove the warning.
[ERROR   ] Failed to read the virtual function for module: opt-1
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/salt/loader.py", line 1775, in _process_virtual
    virtual
  File "/usr/local/lib/python3.6/site-packages/salt/utils/versions.py", line 138, in warn_until
    salt_version=_version_.formatted_version
RuntimeError: The warning triggered on filename '/usr/local/lib/python3.6/site-packages/salt/loader.py', line number 1775, is supposed to be shown until version 2014.1.0 (Hydrogen) is released. Current version is now 2018.3.1 (Oxygen). Please remove the warning.
[INFO    ] User root Executing command saltutil.find_job with jid 20181001141330763305
[DEBUG   ] Command details {'fun': 'saltutil.find_job', 'arg': ['20181001141315539875'], 'tgt': ['GrueWorld'], 'jid': '20181001141330763305', 'ret': '', 'tgt_type': 'list', 'user': 'root'}
[INFO    ] Starting a new job with PID 65823
[DEBUG   ] LazyLoaded saltutil.find_job
[DEBUG   ] LazyLoaded direct_call.execute
[DEBUG   ] Minion return retry timer set to 9 seconds (randomized)
[INFO    ] Returning information for job: 20181001141330763305
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
[DEBUG   ] Trying to connect to: tcp://127.0.0.1:4506
[DEBUG   ] minion return: {'success': True, 'return': {'pid': 65817, 'fun': 'pkg.install', 'arg': ['vim'], 'tgt': '*', 'jid': '20181001141315539875', 'ret': '', 'tgt_type': 'glob', 'user': 'root'}, 'retcode': 0, 'jid': '20181001141330763305', 'fun': 'saltutil.find_job', 'fun_args': ['20181001141315539875']}
[INFO    ] User root Executing command saltutil.find_job with jid 20181001141340991657
[DEBUG   ] Command details {'fun': 'saltutil.find_job', 'arg': ['20181001141315539875'], 'tgt': ['GrueWorld'], 'jid': '20181001141340991657', 'ret': '', 'tgt_type': 'list', 'user': 'root'}
[INFO    ] Starting a new job with PID 65826
[DEBUG   ] LazyLoaded saltutil.find_job
[DEBUG   ] LazyLoaded direct_call.execute
[DEBUG   ] Minion return retry timer set to 10 seconds (randomized)
[INFO    ] Returning information for job: 20181001141340991657
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
[DEBUG   ] Trying to connect to: tcp://127.0.0.1:4506
[DEBUG   ] minion return: {'success': True, 'return': {'pid': 65817, 'fun': 'pkg.install', 'arg': ['vim'], 'tgt': '*', 'jid': '20181001141315539875', 'ret': '', 'tgt_type': 'glob', 'user': 'root'}, 'retcode': 0, 'jid': '20181001141340991657', 'fun': 'saltutil.find_job', 'fun_args': ['20181001141315539875']}
[INFO    ] User root Executing command saltutil.find_job with jid 20181001141351220183
[DEBUG   ] Command details {'fun': 'saltutil.find_job', 'arg': ['20181001141315539875'], 'tgt': ['GrueWorld'], 'jid': '20181001141351220183', 'ret': '', 'tgt_type': 'list', 'user': 'root'}
[INFO    ] Starting a new job with PID 65829
[DEBUG   ] LazyLoaded saltutil.find_job
[DEBUG   ] LazyLoaded direct_call.execute
[DEBUG   ] Minion return retry timer set to 10 seconds (randomized)
[INFO    ] Returning information for job: 20181001141351220183
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
[DEBUG   ] Trying to connect to: tcp://127.0.0.1:4506
[DEBUG   ] minion return: {'success': True, 'return': {'pid': 65817, 'fun': 'pkg.install', 'arg': ['vim'], 'tgt': '*', 'jid': '20181001141315539875', 'ret': '', 'tgt_type': 'glob', 'user': 'root'}, 'retcode': 0, 'jid': '20181001141351220183', 'fun': 'saltutil.find_job', 'fun_args': ['20181001141315539875']}
[INFO    ] User root Executing command saltutil.find_job with jid 20181001141401422011
[DEBUG   ] Command details {'fun': 'saltutil.find_job', 'arg': ['20181001141315539875'], 'tgt': ['GrueWorld'], 'jid': '20181001141401422011', 'ret': '', 'tgt_type': 'list', 'user': 'root'}
[INFO    ] Starting a new job with PID 65832
[DEBUG   ] LazyLoaded saltutil.find_job
[DEBUG   ] LazyLoaded direct_call.execute
[DEBUG   ] Minion return retry timer set to 9 seconds (randomized)
[INFO    ] Returning information for job: 20181001141401422011
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
[DEBUG   ] Trying to connect to: tcp://127.0.0.1:4506
[DEBUG   ] minion return: {'success': True, 'return': {'pid': 65817, 'fun': 'pkg.install', 'arg': ['vim'], 'tgt': '*', 'jid': '20181001141315539875', 'ret': '', 'tgt_type': 'glob', 'user': 'root'}, 'retcode': 0, 'jid': '20181001141401422011', 'fun': 'saltutil.find_job', 'fun_args': ['20181001141315539875']}
[INFO    ] User root Executing command saltutil.find_job with jid 20181001141411567698
[DEBUG   ] Command details {'fun': 'saltutil.find_job', 'arg': ['20181001141315539875'], 'tgt': ['GrueWorld'], 'jid': '20181001141411567698', 'ret': '', 'tgt_type': 'list', 'user': 'root'}
[INFO    ] Starting a new job with PID 65835
[DEBUG   ] LazyLoaded saltutil.find_job
[DEBUG   ] LazyLoaded direct_call.execute
[DEBUG   ] Minion return retry timer set to 8 seconds (randomized)
[INFO    ] Returning information for job: 20181001141411567698
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
[DEBUG   ] Trying to connect to: tcp://127.0.0.1:4506
[DEBUG   ] minion return: {'success': True, 'return': {'pid': 65817, 'fun': 'pkg.install', 'arg': ['vim'], 'tgt': '*', 'jid': '20181001141315539875', 'ret': '', 'tgt_type': 'glob', 'user': 'root'}, 'retcode': 0, 'jid': '20181001141411567698', 'fun': 'saltutil.find_job', 'fun_args': ['20181001141315539875']}
[INFO    ] User root Executing command saltutil.find_job with jid 20181001141421876705
[DEBUG   ] Command details {'fun': 'saltutil.find_job', 'arg': ['20181001141315539875'], 'tgt': ['GrueWorld'], 'jid': '20181001141421876705', 'ret': '', 'tgt_type': 'list', 'user': 'root'}
[INFO    ] Starting a new job with PID 65882
[DEBUG   ] LazyLoaded saltutil.find_job
[DEBUG   ] LazyLoaded direct_call.execute
[DEBUG   ] Minion return retry timer set to 8 seconds (randomized)
[INFO    ] Returning information for job: 20181001141421876705
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
[DEBUG   ] Trying to connect to: tcp://127.0.0.1:4506
[DEBUG   ] minion return: {'success': True, 'return': {'pid': 65817, 'fun': 'pkg.install', 'arg': ['vim'], 'tgt': '*', 'jid': '20181001141315539875', 'ret': '', 'tgt_type': 'glob', 'user': 'root'}, 'retcode': 0, 'jid': '20181001141421876705', 'fun': 'saltutil.find_job', 'fun_args': ['20181001141315539875']}
[DEBUG   ] DSC: Only available on Windows systems
[DEBUG   ] DSC: Only available on Windows systems
[DEBUG   ] Module PSGet: Only available on Windows systems
[DEBUG   ] Module PSGet: Only available on Windows systems
[DEBUG   ] Could not LazyLoad pkg.install: 'pkg.install' is not available.
[DEBUG   ] Minion return retry timer set to 5 seconds (randomized)
[INFO    ] Returning information for job: 20181001141315539875
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/usr/local/etc/salt/pki/minion', 'GrueWorld', 'tcp://127.0.0.1:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
[DEBUG   ] Trying to connect to: tcp://127.0.0.1:4506
[DEBUG   ] minion return: {'success': False, 'return': "'pkg.install' is not available.", 'retcode': 254, 'out': 'nested', 'jid': '20181001141315539875', 'fun': 'pkg.install', 'fun_args': ['vim']}
```

### Setup

The Master and the single Minion are running on the same machine.

Next I attempted to run the 'pkg.install' module to install vim as a test run.

For info regarding the system that salt is running on, my output for `salt '*' grains.items` is:
```
GrueWorld:
    ----------
    SSDs:
    biosreleasedate:
        01/01/2007
    biosversion:
        0.5.1
    cpu_flags:
        - FPU
        - DE
        - PSE
        - TSC
        - MSR
        - PAE
        - MCE
        - CX8
        - APIC
        - SEP
        - MTRR
        - PGE
        - MCA
        - CMOV
        - PAT
        - PSE36
        - MMX
        - FXSR
        - SSE
        - SSE2
        - SSE3
        - CX16
        - HV
        - SYSCALL
        - NX
        - LM
        - LAHF
        - ABM
    cpu_model:
        QEMU Virtual CPU version (cpu64-rhel6)
    cpuarch:
        amd64
    disks:
        ----------
        ada0:
            ----------
            Mediasize:
                21474836480 (20G)
            Sectorsize:
                512
            descr:
                QEMU HARDDISK
            device_model:
                QEMU HARDDISK
            ident:
                QM00001
            media_RPM:
                None
            rotationrate:
                None
            serial_number:
                QM00001
    dns:
        ----------
        domain:
        ip4_nameservers:
            - 8.8.8.8
            - 8.8.4.4
        ip6_nameservers:
        nameservers:
            - 8.8.8.8
            - 8.8.4.4
        options:
        search:
        sortlist:
    domain:
    fc_wwn:
        False
    fqdn:
        GrueWorld
    fqdn_ip4:
    fqdn_ip6:
    gid:
        0
    gpus:
    groupname:
        wheel
    host:
        GrueWorld
    hwaddr_interfaces:
        ----------
        vtnet0:
            00:16:3c:17:d7:8f
    id:
        GrueWorld
    interfaces:
        None
    ip4_interfaces:
        ----------
        lo0:
            - 127.0.0.1
        lo1:
        vtnet0:
            - 176.56.236.234
    ip6_interfaces:
        ----------
        lo0:
            - ::1
            - fe80::1
        lo1:
        vtnet0:
    ip_interfaces:
        ----------
        lo0:
            - 127.0.0.1
            - ::1
            - fe80::1
        lo1:
        vtnet0:
            - 176.56.236.234
    ipv4:
        - 127.0.0.1
        - 176.56.236.234
    ipv6:
        - ::1
        - fe80::1
    iscsi_iqn:
        False
    kernel:
        unknown
    kernelrelease:
        11.2-RELEASE
    kernelversion:
        FreeBSD 11.2-RELEASE #0 r335510: Fri Jun 22 04:32:14 UTC 2018     root@releng2.nyi.freebsd.org:/usr/obj/usr/src/sys/GENERIC
    locale_info:
        ----------
        defaultencoding:
            None
        defaultlanguage:
            None
        detectedencoding:
            US-ASCII
    localhost:
        GrueWorld
    manufacturer:
        Red Hat
    master:
        127.0.0.1
    mem_total:
        989
    model:
        None
    nodename:
        GrueWorld
    num_cpus:
        2
    num_gpus:
        0
    optional_args:
        ----------
    os:
        None
    os_family:
        proxy
    osarch:
        amd64
    osfinger:
        FreeBSD-11
    osmajorrelease:
        11
    osrelease:
        proxy
    osrelease_info:
        - 11
        - 2
    path:
        /sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/root/bin
    pid:
        59250
    product:
        HUE
    productname:
        KVM
    ps:
        ps auxwww
    pythonexecutable:
        /usr/local/bin/python3.6
    pythonpath:
        - /usr/local/bin
        - /usr/local/lib/python36.zip
        - /usr/local/lib/python3.6
        - /usr/local/lib/python3.6/lib-dynload
        - /usr/local/lib/python3.6/site-packages
    pythonversion:
        - 3
        - 6
        - 6
        - final
        - 0
    saltpath:
        /usr/local/lib/python3.6/site-packages/salt/grains
    saltversion:
        2018.3.1
    saltversioninfo:
        - 2018
        - 3
        - 1
        - 0
    serial:
        None
    serialnumber:
        kenv: unable to get smbios.system.serial
    server_id:
        1565285772
    shell:
        /bin/csh
    swap_total:
        1023
    uid:
        0
    uptime:
        None
    username:
        root
    uuid:
        29be8eb5-a596-4b2b-83e1-4f5dd42ea2d1
    vendor:
        None
    version:
        None
    virtual:
        kvm
    zfs_feature_flags:
        True
    zfs_support:
        False
    zmqversion:
        4.2.3
```

### Steps to Reproduce Issue
Simply attempt to use the pkg.install module to install any package, in this case vim. (formerly known as pkgng.install and freebsdpkg.install)
`salt '*' pkg.install vim`

### Versions Report

`salt --versions-report`:

```
Salt Version:
           Salt: 2018.3.1
 
Dependency Versions:
           cffi: 1.11.5
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.6 (default, Sep 27 2018, 09:04:39)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.3
 
System Versions:
           dist:   
         locale: US-ASCII
        machine: amd64
        release: 11.2-RELEASE
         system: FreeBSD
        version: Not Installed
```

**Comment by SamuelGogarty** [comment]
It seems a similiar bug was reported here: https://github.com/saltstack/salt/issues/47006

A work around for this is to run the following: 

`find /usr/local/lib/python3.6/site-packages/salt -name '*opt-1*' -delete`

Restarting my Master and Minion, waiting a couple minutes and then 
attempting to use the `pkg.install `module did the trick for me.

**Comment by Ch3LL** [comment]
it looks like in that issue that a fix is in 2018.3.2 as well. If you upgrade does it also work for you?

**Comment by SamuelGogarty** [comment]
> it looks like in that issue that a fix is in 2018.3.2 as well. If you upgrade does it also work for you?
It seem's like my server's release of FreeBSD is stuck with 2018.3.1 at the moment, although I have another machine running a testing FreeBSD release in which 2018.3.2 is available. I'll post my results after I test it.
